python/cudf_polars/cudf_polars/experimental/base.py

```diff
@@ -20,19 +20,23 @@
 class PartitionInfo:
     """Partitioning information."""
 
-    __slots__ = ("count", "partitioned_on")
+    __slots__ = ("count", "metadata", "partitioned_on")
     count: int
     """Partition count."""
     partitioned_on: tuple[NamedExpr, ...]
     """Columns the data is hash-partitioned on."""
+    metadata: dict[str, Any] | None
+    """Optional metadata."""
```

Comment:
We don't actually need arbitrary metadata. This attribute is only used to track the `ScanPartitionPlan` needed for `Scan` nodes.  This is because it no-longer makes much sense to generate a distinct `Scan` node for every individual partition during the lowering phase. This made sense when each `Scan` node could be mapped to a distinct task. However, we now want a single streaming node to generate all chunks "belonging" to the local rank.

My current thinking is that the `ScanPartitionPlan` should still be collected at lowering time (when statistics are available), but they should not be used to modify the IR-graph. Instead, the "partitioning plan" should be attached to the `PartitionInfo` in some way.